packages/gatsby-transformer-remark/src/extend-node-type.js

```diff
@@ -139,21 +152,71 @@ module.exports = function remarkExtendNodeType(
       }
     }
 
-    async function getAST(markdownNode) {
-      const cacheKey = astCacheKey(markdownNode)
+    async function addCacheNodeDependency(markdownNode, depNode) {
+      const digestCacheKey = getCacheKey(CACHE_TYPE_NODE_DEP_DIGEST, depNode)
+      await cache.set(digestCacheKey, depNode.internal.contentDigest)
+
+      const depCacheKey = getCacheKey(CACHE_TYPE_NODE_DEPS, markdownNode)
+      const dependencies = await cache.get(depCacheKey)
+
+      if (!dependencies) {
+        await cache.set(depCacheKey, {
+          [depNode.id]: depNode.internal.contentDigest,
+        })
+      } else if (dependencies.indexOf(depNode.id) < 0) {
+        dependencies[depNode.id] = depNode.internal.contentDigest
+        await cache.set(depCacheKey, dependencies)
+      }
+    }
+
+    async function getCacheWithNodeDependencyValidation(
+      cacheKey,
+      markdownNode
+    ) {
+      const depCacheKey = getCacheKey(CACHE_TYPE_NODE_DEPS, markdownNode)
+      const dependencies = await cache.get(depCacheKey)
+
+      // TODO don't return if there are actually no deps for a certain node
+      if (!dependencies) {
+        return undefined
+      }
+
+      const nodeIds = Object.keys(dependencies)
+
+      for (let i = 0; i < nodeIds.length; i++) {
+        // TODO improve this key-fetch method here, generating an object just to conform isn't great
+        const node = { id: nodeIds[i] }
+        const digestCacheKey = getCacheKey(CACHE_TYPE_NODE_DEP_DIGEST, node)
+        const prevDepDigest = await cache.get(digestCacheKey)
+        const curDepDigest = dependencies[nodeIds[i]]
```

Comment:
I don't quite get how the `cur` variant should work - it still uses something cached to get digest, but I think we do need to get current node with that `id` from gatsby datastore and not use cache for that, because cache won't necesairly have all things updated?

Reply:
You're right. What was here before was broken from a previous iteration. Should be updated now!